src/Lecture1.hs

```diff
@@ -119,4 +145,16 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+lowerAndGreater :: (Num a, Ord a, Show a) => a -> [a] -> String
+lowerAndGreater n list =
+  (show n) ++ " is greater than " ++ (show greater) ++
+  " elements and lower than " ++ (show lower) ++ " elements"
+  where
+    -- this recurses through the list, incrementing # nums greater + lower
+    accumulator :: (Num a, Ord a) => a -> a -> a -> [a] -> (a, a)
+    accumulator numGr numLw _ [] = (numGr, numLw)
+    accumulator numGr numLw targNum theList
+      | targNum > head theList = accumulator (numGr+1) numLw targNum (tail theList)
+      | targNum < head theList = accumulator numGr (numLw+1) targNum (tail theList)
+      | otherwise = accumulator (numGr) numLw targNum (tail theList)
+    (greater, lower) = accumulator 0 0 n list
```

Comment:
I see that the number `n` that you pass in recursive calls doesn't change. You always pass the same number. In that case, you can remove this extra argument and use `n` from the function instead. It's possible because functions in `where` can see all the arguments of the top-level function 🔍